src/openssl.c

```diff
@@ -405,28 +705,50 @@ _libssh2_ecdsa_curve_name_with_octal_new(libssh2_ecdsa_ctx ** ec_ctx,
 
         if(ec_ctx)
             *ec_ctx = ec_key;
+#endif
     }
 
     return (ret == 1) ? 0 : -1;
 }
 
+#ifdef USE_OPENSSL_3
+#define LIBSSH2_ECDSA_VERIFY(digest_type)                           \
+{                                                                   \
+    unsigned char hash[SHA##digest_type##_DIGEST_LENGTH];           \
+    libssh2_sha##digest_type(m, m_len, hash);                       \
+    ret = EVP_PKEY_verify_init(ctx);                                \
+    if(ret > 0) {                                                   \
+        ret = EVP_PKEY_verify(ctx, der, der_len, hash,              \
+                              SHA##digest_type##_DIGEST_LENGTH);    \
+    }                                                               \
+                                                                    \
+}
```

Comment:
I'd suggest enclosing this also in a `do {} while(0)`. This way we can keep the terminating `;` in the macro calls.

Reply:
Good call. This was one of those things were I was just trying to get it to work, so I was editing the macro, and wanted to keep it as simple as possible, then forgot to come back and wrap it when I was finished.

I've pushed a new commit to do as you suggested.